engine/tests/phpunit/Elgg/FormsServiceTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Elgg;
+
+/**
+ * @group FormsService
+ */
+class FormsServiceTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @var ViewsService
+	 */
+	protected $views;
+
+	/**
+	 * @var string
+	 */
+	protected $viewsDir;
+
+	public function setUp() {
+		$this->viewsDir = dirname(dirname(__FILE__)) . "/test_files/views";
+
+		$this->hooks = new PluginHooksService();
+		$logger = $this->getMock('\Elgg\Logger', array(), array(), '', false);
+
+		$this->views = new ViewsService($this->hooks, $logger);
+		$this->views->autoregisterViews('', "$this->viewsDir/default", 'default');
+
+		_elgg_services()->setValue('views', $this->views);
+		_elgg_services()->setValue('hooks', $this->hooks);
```

Comment:
Yeah, a way to refresh the whole SP is the only sane way. The logger and hooks already do have backup/restore facilities; we should just use the SP ones instead of making mocks here.